backend/routes/chatRouter.js

```diff
@@ -1,33 +1,80 @@
 const express = require("express");
 const path = require("path");
 const chatRouter = express.Router();
+const Chat = require("../models/Chat.js");
+
+// Join a chat
+chatRouter.get("/join/:chat_id", async (req, res) => {
+	console.log("GET request recieved");
+    const chatId = req.params.chat_id;
+
+    try {
+		//Check if a chat with the given PIN exists
+		console.log("Attempting to find chat");
+        const chat = await Chat.findOne({ pin: chatId });
+		console.log("Chat found");
+
+        if (chat) {
+			//Respond true if the chatroom exists
+			res.json({ exists: true });
+        } else {
+            res.status(404).json({ exists: false, message: "Chatroom not found." });
+        }
+    } catch (error) {
+        console.error("Error checking chatroom existence:", error);
+        res.status(500).json({ message: "An error occurred while checking the chatroom." });
+    }
+});
+
+//Utility function to generate PIN
+function generateRandomPin() {
+    return Math.floor(10000000 + Math.random() * 90000000);
+}
+
+// Create a new chat
+chatRouter.post("/create", async (req, res) => {
+	console.log("POST request recieved");
+    const pin = generateRandomPin();
+    const chatName = "Anonymous Chat";
+
+    try {
+		console.log("Attempting to create new Chat")
+        // Create a new chat instance
+        const newChat = new Chat({
+            name: chatName,
+            pin: pin,
+            users: [],
```

Comment:
You can probably create the guest user by doing what I stated above and add them here now

Reply:
Solved